src/webgl/dataLoader.js

```diff
@@ -0,0 +1,542 @@
+
+(function($) {
+
+/**
+ * IDataLoader conforms to a specific texture type and WebGL version.
+ * It provides API for uniform handling of textures:
+ *  - texture loading
+ *  - GLSL texture handling
+ */
+$.WebGLModule.IDataLoader = class {
```

Comment:
@Aiosa What does the `I` stand for here?

Reply:
It indicates this class is an 'interface', meant for extension, not an instantiation, Honestly, I think the data loader can get removed once we finish https://github.com/openseadragon/openseadragon/pull/2407 since then we can just say `conversion image -> TEXTURE_2D happens like this` directly on the whole openseadragon level